drivers/storage/portworx/deployment.go

```diff
@@ -1120,7 +1191,95 @@ func (t *template) getCSIVolumeInfoList() []volumeInfo {
 	return volumeInfoList
 }
 
+func (t *template) getTelemetryVolumeInfoList() []volumeInfo {
+	if t.cluster.Spec.Telemetry != nil && t.cluster.Spec.Telemetry.Enabled {
+
+		configVolume := volumeInfo{
+			name:      "ccm-config",
+			mountPath: "/etc/ccm",
+			configMapType: &v1.ConfigMapVolumeSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: component.TelemetryConfigMapName,
+				},
+				Items: []v1.KeyToPath{
+					{
+						Key:  component.TelemetryPropertiesFilename,
+						Path: component.TelemetryPropertiesFilename,
+					},
+				},
+			},
+		}
+
+		// reference location from config map only if the default was overridden using annotations
+		if location, present := t.cluster.Annotations[pxutil.AnnotationTelemetryArcusLocation]; present && len(location) > 0 {
+			configVolume.configMapType.Items = append(configVolume.configMapType.Items, v1.KeyToPath{
+				Key:  component.TelemetryArcusLocationFilename,
+				Path: component.TelemetryArcusLocationFilename,
+			})
+		}
+		volumeInfoList := []volumeInfo{
+			{
+				name:      "diagsdump",
+				hostPath:  "/var/cores",
+				mountPath: "/var/cores",
+				pks: &pksVolumeInfo{
+					hostPath: "/var/vcap/store/cores",
+				},
+			},
+			{
+				name:      "etcpwx",
+				hostPath:  "/etc/pwx",
+				mountPath: "/etc/pwx",
+				pks: &pksVolumeInfo{
+					hostPath: "/var/vcap/store/etc/pwx",
+				},
+			},
+			{
+				name: "pxlogs",
+				pks: &pksVolumeInfo{
+					hostPath:  "/var/vcap/store/lib/osd/log",
+					mountPath: "/var/lib/osd/log",
+				},
+			},
+			{
+				name:      "journalmount1",
+				hostPath:  "/var/run/log",
+				mountPath: "/var/run/log",
+				readOnly:  true,
+			},
+			{
+				name:      "journalmount2",
+				hostPath:  "/var/log",
+				mountPath: "/var/log",
+				readOnly:  true,
+			},
```

Comment:
For the common volumes, can we extract it in a common place? That will avoid duplication and chance of error, if someone modifies the volume only in one place.

Reply:
done